zanata-war/src/main/java/org/zanata/service/impl/ActivityServiceImpl.java

```diff
@@ -184,24 +185,34 @@ public Object getEntity(EntityType entityType, long entityId) {
     public void logTextFlowStateUpdate(@Observes(during = TransactionPhase.AFTER_SUCCESS) TextFlowTargetStateEvent event_) {
         // workaround for https://issues.jboss.org/browse/WELD-2019
         final TextFlowTargetStateEvent event = event_;
-        Long actorId = event.getActorId();
+
+        Long actorId = event.getKey().getActorId();
+
         if (actorId != null) {
             Lock lock = activityLockManager.getLock(actorId);
             lock.lock();
             try {
-                transactionUtil.run(() -> {
-                    HTextFlowTarget target =
-                            textFlowTargetDAO.findById(event.getTextFlowTargetId(),
-                                    false);
-                    HDocument document = documentDAO.getById(event.getDocumentId());
-                    ActivityType activityType =
-                            event.getNewState().isReviewed() ? ActivityType.REVIEWED_TRANSLATION
-                                    : ActivityType.UPDATE_TRANSLATION;
-
-                    logActivityAlreadyLocked(actorId,
-                            document.getProjectIteration(), target, activityType,
-                            target.getTextFlow().getWordCount().intValue());
-                });
+                HDocument document =
+                    documentDAO.getById(event.getKey().getDocumentId());
+
+                for (TextFlowTargetStateEvent.TextFlowTargetState state : event
+                    .getStates()) {
+                    transactionUtil.run(() -> {
+                        HTextFlowTarget target =
+                            textFlowTargetDAO.findById(
+                                state.getTextFlowTargetId(), false);
+                        ActivityType activityType =
+                            state.getNewState().isReviewed()
+                                ? ActivityType.REVIEWED_TRANSLATION
+                                : ActivityType.UPDATE_TRANSLATION;
+
+                        logActivityAlreadyLocked(actorId,
```

Comment:
Sounds good.

On 15 April 2016 at 15:27, Alex Eng <notifications@github.com> wrote:

> In
> zanata-war/src/main/java/org/zanata/service/impl/ActivityServiceImpl.java
> <https://github.com/zanata/zanata-server/pull/1138#discussion_r59828360>:
>
> > -                });
> > +                HDocument document =
> > +                    documentDAO.getById(event.getKey().getDocumentId());
> > +
> > +                for (TextFlowTargetStateEvent.TextFlowTargetState state : event
> > +                    .getStates()) {
> > +                    transactionUtil.run(() -> {
> > +                        HTextFlowTarget target =
> > +                            textFlowTargetDAO.findById(
> > +                                state.getTextFlowTargetId(), false);
> > +                        ActivityType activityType =
> > +                            state.getNewState().isReviewed()
> > +                                ? ActivityType.REVIEWED_TRANSLATION
> > +                                : ActivityType.UPDATE_TRANSLATION;
> > +
> > +                        logActivityAlreadyLocked(actorId,
>
> But then translation and review is 2 different activity type. We can try
> to aggregate these 2 types of activity, total up the word counts and fire 2
> event.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1138/files/462a372aa89c14a002268b729deeb53d52a19f33#r59828360>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat